### PR TITLE
Query Tuning beta docs

### PR DESCRIPTION
**Pull request by lfittl**
Adds the beta docs, and reorders the docs navigation items a bit to match in-app order.

## TODO

* [x] Apply review feedback
* [x] Refresh screenshots
* [x] Rework query optimization example cases (these are still a rough draft, and don't read well / are hard to follow)

### CODE REVIEW

File: query-tuning/index.mdx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+<DocsBetaFeature feature="Query Tuning" />
+
+## Getting Started
+
+Query Tuning provides a unified interface and workflow for optimizing slow Postgres queries.
+
+The Query Tuning Workbooks feature lets you streamline recording of related `EXPLAIN ANALYZE`
+outputs and enables easy, secure sharing them with your team. Each workbook contains a
+baseline query and one or more variants.
```
Comment (msakrejda) [inline query-tuning/index.mdx:17]:
```suggestion
The Query Tuning Workbooks feature lets you organize related `EXPLAIN ANALYZE`
outputs and share them easily and securely with your team. Each workbook contains a
baseline query documenting a query's current performance, and one or more variants.
```

File: query-tuning/index.mdx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+Variants are a modification to the original query that aims to optimize it by changing planner
```
Comment (msakrejda) [inline query-tuning/index.mdx:19]:
```suggestion
Variants are modifications to the original query that aim to optimize it by changing planner
```
or
```suggestion
A variant is a modification to the original query that aims to optimize it by changing planner
```

File: query-tuning/index.mdx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+settings, adding a `pg_hint_plan` hint, or rewriting the query. You can compare and analyze
+variants with clear visibility into changes and their effects on performance.
+
+The parameter sets feature makes it easy to test performance for different bind parameter values
+across the baseline and variants, making sure that optimizing for one user's experience does not
+slow down other cases.
```
Comment (msakrejda) [inline query-tuning/index.mdx:25]: I think it may not be obvious to everyone that parameters can affect performance, so we may want to lead with that:
```suggestion
Different parameter values (or different constant values) can lead to varying performance or even a different plan for a specific query.
The parameter sets feature makes it easy to test performance for different bind parameter values
across the baseline and variants, making sure that optimizing for one user's experience does not
slow down other cases.
```

File: query-tuning/index.mdx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+Query Tuning workbooks automatically convert positional parameters to named parameters, e.g. turning
+`user_id = $1` into `user_id = $user_id`. This makes it easy to test rewrites of a query, or test
+individual portions of a query.
+
+### Query Tuning Features
```
Comment (msakrejda) [inline query-tuning/index.mdx:31]: This feels a bit more like marketing than like documentation. I can see why we want to list it, but it kind of seems out of place here...

File: query-tuning/query-optimization-examples.mdx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+When the query planner chooses nested loops for joins where other strategies might be
+more efficient, Query Tuning can help identify and resolve the issue.
+
+### Detection
+
+Query Tuning automatically flags cases where:
+
+* Nested Loop joins show actual rows significantly higher than estimated
+* Inner side operations consume substantial time
```
Comment (msakrejda) [inline query-tuning/query-optimization-examples.mdx:29]: It's not clear if we're talking about AND or OR for these two...

File: query-tuning/collector-workflow.mdx [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+---
+
+import CopyCodeBlock from '../components/CopyCodeBlock'
+import DocsBetaFeature from '../components/DocsBetaFeature'
+import { explainAnalyzeHelper } from '../components/helperSQL'
+
+<DocsBetaFeature feature="Query Tuning" />
+
+The Collector Workflow in Query Tuning workbooks optionally allows users with the [Tune Queries permission]()
+to execute arbitrary EXPLAIN ANALYZE commands directly from the pganalyze user interface, avoiding tedious
```
Comment (keiko713) [inline query-tuning/collector-workflow.mdx:14]: (nitpicking, fine as is too) feels like "avoiding tedious copy and pasting" contains some emotional touch, where I'd like the docs to explain the fact. How about changing this to something like:

"...from the pganalyze user interface. This eliminates the need for manual copying and pasting of command outputs."

File: query-tuning/collector-workflow.mdx [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+copy and pasting of command outputs. These EXPLAIN ANALYZE commands are executed
+through the pganalyze Collector.
```
Comment (keiko713) [inline query-tuning/collector-workflow.mdx:16]:
```suggestion
through the pganalyze collector.
```

We usually use the lowercase for the collector.

File: query-tuning/howto_setup_1_review_query.png
Comment (keiko713) [inline query-tuning/howto_setup_1_review_query.png:1]: For some reason, this png is rendered super strange in the review app: https://pganalyze-we-query-tuni-nl5vxq.herokuapp.com/docs/query-tuning/how-to-use-workbooks#review-your-query

File: query-tuning/index.mdx [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+## How to enable Query Tuning
+
+### New role and permissions
+
+Query Tuning utilizes a separate user permission, called “Tune Queries”. Only users with
+Admin roles will get this permission assigned automatically. This permission allows
+users to create workbooks and to use all associated Query Tuning features.
+
+Users without the permission are able to view existing workbooks, including parameter
+information, but cannot create new workbooks, variants, or record new plans.
```
Comment (keiko713) [inline query-tuning/index.mdx:57]: what is "record new plans"?

File: query-tuning/index.mdx [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+## Limitations
+
+* Workbooks are currently not suited for optimizing queries invoked inside of functions
+* Workbooks are currently not suited for testing different index choices by doing CREATE INDEX
+* Data modification (DML) queries are not supported
+* All query executions are limited to 60 seconds (this limit will be configurable in the future)
+* The Collector Workflow requires the use of a function not owned by superuser
+  * If your database provider does not allow creating separate roles, you cannot use the Collector Workflow
+
+[Learn how to navigate and use workbooks next](how-to-use-workbooks)
```
Comment (keiko713) [inline query-tuning/index.mdx:80]: This link is not working well. I think we usually do something like `[](/docs/query-tuning/how-to-use-workbooks)`

File: query-tuning/query-optimization-examples.mdx [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+
+Query optimization often involves recognizing and addressing common patterns that lead
+to suboptimal performance. While seasoned database engineers may spot these patterns
+through experience, Query Tuning helps identify and solve them systematically. Each
+pattern has specific indicators that suggest its presence, methods to validate the
+issue, and proven approaches to resolve it.
+
+We'll explore three common patterns that often impact query performance.
+
+## Inefficient Nested Loops Pattern
```
Comment (keiko713) [inline query-tuning/query-optimization-examples.mdx:19]: It would be nice to have a ToC (list) in the beginning of this page, so that people can jump to these 3 patterns

File: query-tuning/query-optimization-examples.mdx [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+* Merge Joins where index sort order appears to take precedence over filtering
+* Cases where parameterized index scans might be beneficial
+
+### Investigation Process
+
+1. Start with baseline plan analysis:
+   * Upload your current EXPLAIN plan
+   * Review highlighted index usage patterns
+2. Check scan efficiency metrics
+   * Create test variants:
```
Comment (keiko713) [inline query-tuning/query-optimization-examples.mdx:64]: typo? there is a tailing :

File: query-tuning/query-optimization-examples.mdx [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+   * ORDER BY col+0 modification
+   * enable_incremental_sort = off setting
+3. Compare performance across variants
+
+### Benchmarking Solutions
+
+1. Test different statistics target values
+2. Compare plans with and without incremental sort
+3. Verify performance across different LIMIT values
+4. Document the most effective approach
```
Comment (keiko713) [inline query-tuning/query-optimization-examples.mdx:103]: We can have a section of "reference" and have a link to 5mins of postgres potentially

File: query-tuning/collector-workflow.mdx [hunk 13]
```diff
@@ -0,0 +1,128 @@
...
+backlink_title: 'Query Tuning'
+---
+
+import CopyCodeBlock from '../components/CopyCodeBlock'
+import DocsBetaFeature from '../components/DocsBetaFeature'
+import { explainAnalyzeHelper } from '../components/helperSQL'
+
+<DocsBetaFeature feature="Query Tuning" />
+
+The Collector Workflow in Query Tuning workbooks optionally allows users with the [Tune Queries permission]()
```
Comment (keiko713) [inline query-tuning/collector-workflow.mdx:13]: I think you can fill `/docs/accounts/permissions#tune-queries` already here as it'll be the link

**Review by keiko713 — APPROVED** [review]
Lots of small things but looks good to me 🎉

File: query-tuning/index.mdx (hunk 4 above)
Comment (lfittl) [reply query-tuning/index.mdx:31]: Yeah, this was from the original docs draft and I also had that feedback then - let me see if I can cut it down / rephrase / remove.

File: query-tuning/howto_setup_1_review_query.png
Comment (lfittl) [reply query-tuning/howto_setup_1_review_query.png:1]: Hmm, yeah, I think that is Gatsby's image optimization pipeline trying to be smart.

File: query-tuning/collector-workflow.mdx (hunk 6 above)
Comment (lfittl) [reply query-tuning/collector-workflow.mdx:14]: +1 for "I'd like the docs to explain the fact.", thanks for calling that out.

File: query-tuning/index.mdx (hunk 8 above)
Comment (lfittl) [reply query-tuning/index.mdx:58]: Basically that they can't upload plans. I'll reword to "use the collector workflow", since that's the main thing that matters.

File: query-tuning/index.mdx (hunk 4 above)
Comment (lfittl) [reply query-tuning/index.mdx:32]: I've decided to take most of this out since it was actually duplicative of the prior section, with the exception of plan comparison, which I've edited into the above text.

File: query-tuning/query-optimization-examples.mdx (hunk 12 above)
Comment (lfittl) [reply query-tuning/query-optimization-examples.mdx:103]: I think this is generally a good idea, however we don't have 5mins episodes for each of these it seems, so for now I've added two inline links to specific episodes, to be revised in the future.

File: query-tuning/howto_setup_1_review_query.png
Comment (lfittl) [reply query-tuning/howto_setup_1_review_query.png:1]: Looks like that's still happening even with the updated screenshots. I'll make a TODO item for us to fix this separately.
